lib/database/models/coupon.dart

```diff
@@ -0,0 +1,18 @@
+import 'package:holdam_mobile/database/models/user.dart';
+import 'package:objectbox/objectbox.dart';
+
+import 'promo.dart' show Promo;
+import 'user.dart' show User;
+
+@Entity()
+class Coupon {
+  @Id()
+  int id = 0;
```

Comment:
Why the is initialized? I think it should be autogenerated. That is the functionality of the "id" decorator afaik.

Reply:
https://docs.objectbox.io/entity-annotations#object-ids-id

Reply:
Yes, I thought that too and confused me, but there are a few things about this
1. look at the official documentation here on how they defined the id https://docs.objectbox.io/entity-annotations#objectbox-database-persistence-with-entity-annotations
2. Also when I only leave without initializing gave me this error
<img width="994" alt="image" src="https://github.com/holdamapp/holdam_mobile/assets/65260383/57037a2b-1e81-40a7-a443-0c61e8d65567">

Reply:
I search a little bit and the community initialize the id as 0, object box behind the scene increase 

Look at this tutorial https://tomicriedel.medium.com/94b0586392a5

<img width="989" alt="image" src="https://github.com/holdamapp/holdam_mobile/assets/65260383/eb954765-be12-4f43-9713-50fd402ce9e0">

Reply:
Sounds good!